fedimint-api-client/src/api/mod.rs

```diff
@@ -1173,24 +1234,31 @@ mod iroh {
 
     #[async_trait]
     impl IClientConnection for Connection {
-        async fn request(
-            &self,
-            method: ApiMethod,
-            request: ApiRequestErased,
-        ) -> anyhow::Result<Value> {
-            let json = serde_json::to_vec(&IrohRequest { method, request })?;
+        async fn request(&self, method: ApiMethod, request: ApiRequestErased) -> PeerResult<Value> {
+            let json = serde_json::to_vec(&IrohRequest { method, request })
+                .expect("Serialization to vec can't fail");
 
-            let (mut sink, mut stream) = self.open_bi().await?;
+            let (mut sink, mut stream) = self
+                .open_bi()
+                .await
+                .map_err(|e| PeerError::Transport(e.into()))?;
 
-            sink.write_all(&json).await?;
+            sink.write_all(&json)
+                .await
+                .map_err(|e| PeerError::Transport(e.into()))?;
 
-            sink.finish()?;
+            sink.finish().map_err(|e| PeerError::Transport(e.into()))?;
 
-            let response = stream.read_to_end(1_000_000).await?;
+            let response = stream
+                .read_to_end(1_000_000)
+                .await
+                .map_err(|e| PeerError::Transport(e.into()))?;
 
-            let response = serde_json::from_slice::<Result<Value, ApiError>>(&response)?;
+            // TODO: We should not be serializing Results on the wire
+            let response = serde_json::from_slice::<Result<Value, ApiError>>(&response)
+                .map_err(|e| PeerError::InvalidResponse(e.into()))?;
 
```

Comment:
Todo for later imo: add error to cummunicate "unknown method" etc. from server to client

Reply:
@elsirion @joschisan As is I think the current calling convention is meh here. But this is connected to the question of what serialization format we'd like here etc. so we can argue about it in some deep dive sometime.